python/whylogs/core/resolvers.py

```diff
@@ -99,6 +103,20 @@ def __post_init__(self):
             raise ValueError("MetricSpec: must supply a Metric subclass to MetricSpec")
 
 
+@dataclass
+class ConditionCountMetricSpec(MetricSpec):
+    """
+    Specify a ConditionCountMetric to instantiate.
+    """
+
+    metric: ConditionCountMetric
+    config: Optional[ConditionCountConfig] = None  # omit to use default MetricConfig
+
+    def __post_init__(self):
+        if not issubclass(self.metric, ConditionCountMetric):
+            raise ValueError("ConditionCountMetricSpec: must supply a ConditionCountMetric to MetricSpec")
+
```

Comment:
I was thinking something like (in condition_count_metric.py):
```
@dataclass(init=False)  # I'm not sure that's the exact parameter name
class ConditionCountMetricSpec(MetricSpec):
  __init__(self, conditions: Dict[str, Condition], config: Optional[ConditionCountConfig]=None):
    self.metric = ConditionCountMetric
    self.config = config if config else ConditionCountConfig()
    self.config.conditions.update(conditions)
```
Then you could do
```
date_format_condition = {
    "is_date_format": Condition(Predicate().is_(date_format))
}

date_format_spec = ResolverSpec(
    column_name="last_review",
    metrics=[ConditionCoundMetricSpec(date_format_conditon)],
)
schema = DeclarativeSchema(STANDARD_RESOLVER)
schema.add_resolver(date_format_spec)
```
We could also turn `DatasetSchema::add_resovler()` into more of a factory:
```
def add_resolver_spec(self, column_name: Optional[str]=None, column_type: Optional[Any], metrics: List[MetricSpec]):
  spec = MetricSpec(column_name, column_type, metrics)
  self.add_resolver(spec)
```

So we'd end up with
```
date_format_condition = {
    "is_date_format": Condition(Predicate().is_(date_format))
}

schema = DeclarativeSchema(STANDARD_RESOLVER)
schema.add_resolver_spec("last_review", None, [ConditionCoundMetricSpec(date_format_conditon)])
```
I think this is more a more general simplification.